### PR TITLE
Pic 2290

### PR DESCRIPTION
**Pull request by vb-moj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingProcessor.java [hunk 1]
```diff
@@ -94,11 +94,11 @@ private void updateAndSave(final Hearing hearing) {
     Hearing assignUuids(Hearing hearing) {
         // Apply the new case ID
         final var caseId = UUID.randomUUID().toString();
-        var updatedCase = hearing.withCaseId(caseId).withHearingId(caseId);
+        var updatedHering = hearing.withCaseId(caseId).withHearingId(caseId);
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingProcessor.java:97]: Typo in `updatedHering`

File: src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingExtractorTest.java [hunk 2]
```diff
@@ -89,70 +90,70 @@ void setUp() {
     }
 
     @Test
-    void whenLibraCaseReceived_thenParseAndReturnCase() throws JsonProcessingException {
+    void whenLibraHearingReceived_thenParseAndReturnCase() throws JsonProcessingException {
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingExtractorTest.java:93]: should probably be ..thenParseAndReturn*Hearing*

File: src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSJudicialResultTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public void mapBackToDomain() {
+        final var original = buildJudicialResult();
+        final var actual = CCSJudicialResult.of(original).asDomain();
+
+        assertThat(actual).isEqualTo(original);
+    }
+
+    private JudicialResult buildJudicialResult() {
+        return JudicialResult.builder()
+                .judicialResultType(JudicialResultType.builder().build())
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSJudicialResultTest.java:29]: This needs to populate all available attributes

File: src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSOffenceTest.java [hunk 4]
```diff
@@ -30,6 +32,7 @@ private Offence buildOffence() {
                 .act("act")
                 .sequenceNumber(1)
                 .listNo(30)
+                .judicialResults(Collections.emptyList())
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSOffenceTest.java:35]: I think it would be better to include a single judicial result otherwise it's not really testing that it streams the contents, it could just be defaulting to an empty list.

**Review by jhevans — CHANGES_REQUESTED** [review]
A couple of very minor changes requested but looks good 👍

**Review by jhevans — DISMISSED** [review]
LGTM 👍

**Comment by vb-moj** [comment]
Closing this PR as some re works is needed.
